internal/cli/cli.go

```diff
@@ -35,11 +35,12 @@ func Run(ctx context.Context, io *iostreams.IOStreams, args ...string) int {
 
 	cs := io.ColorScheme()
 
-	defer func() {
-		metrics.FlushPending()
-	}()
+	defer metrics.FlushPending()
 
-	switch _, err := cmd.ExecuteContextC(ctx); {
+	cmd, err := cmd.ExecuteContextC(ctx)
+	defer metrics.RecordCommandFinish(cmd)
```

Comment:
my only note is that maybe we don't want to track failed commands?

Reply:
That's a good call. I've flip flopped back and forth on it, but I think for now I'll restrict this to successful commands only. We don't want timeouts and etc affecting the data.